File: backend/services/group/internal/domain/group.go
package domain

import (
	"errors"
	"time"

	"github.com/google/uuid"
)

var (
	ErrGroupNotFound  = errors.New("group not found")
	ErrMemberNotFound = errors.New("member not found")
)

type Group struct {
	ID          uuid.UUID `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description,omitempty"`
	Currency    string    `json:"currency"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
	Members     []Member  `json:"members"`
}

type Member struct {
	ID       uuid.UUID `json:"id"`
	Name     string    `json:"name"`
	Email    string    `json:"email,omitempty"`
	JoinedAt time.Time `json:"joined_at"`
}

File: frontend/src/types/group.ts
export interface Group {
  id: string;
  name: string;
  description?: string;
  currency: string;
  createdAt: string;
  updatedAt?: string;
  members: Member[];
}

export interface Member {
  id: string;
  name: string;
  email?: string;
  joinedAt: string;
}

export interface CreateGroupInput {
  name: string;
  description?: string;
  currency: string;
  memberNames: string[];
}

export interface Expense {
  id: string;
  groupId: string;
  amount: number;
  description: string;
  paidById: string;
  paidByName: string;
  splitMembers: SplitMember[];
  createdAt: string;
}

export interface SplitMember {
  memberId: string;
  memberName: string;
  amount: number;
}

export interface AddExpenseInput {
  groupId: string;
  amount: number;
  description: string;
  paidById: string;
  splitMemberIds: string[];
}

export interface UpdateExpenseInput {
  expenseId: string;
  amount: number;
  description: string;
  paidById: string;
  splitMemberIds: string[];
}

export interface Settlement {
  fromMemberId: string;
  toMemberId: string;
  amount: number;
  fromName: string;
  toName: string;
}

export interface MemberBalance {
  memberId: string;
  memberName: string;
  balance: number;
}

export interface SettlementResult {
  settlements: Settlement[];
  balances: MemberBalance[];
}

// For calculating settlements
export interface ExpenseInput {
  id: string;
  payerId: string;
  amount: number;
  splitBetween: string[];
}
